spec/features/ingredients/new_ingredient_page_spec.rb

```diff
@@ -0,0 +1,56 @@
+feature 'user creates new ingredient' do
+  before do
+    visit('/ingredients/new')
+  end
+
+  it 'creates new ingredient' do
+    fill_in('ingredient_name', with: 'Example Ingredient')
+    fill_in('ingredient_unit_type', with: 'Type')
+    click_button('Save')
+    current_path.match(ingredients_path)
+    expect(page).to have_content("Example Ingredient Type")
+  end
+
+  context 'when empty params are submitted' do
+    before do
+      fill_in('ingredient_name', with: '')
+      fill_in('ingredient_unit_type', with: '')
+      click_button('Save')
+    end
+
+    it 'displays presence errors' do
+      current_path.match(new_ingredient_path)
+      expect(page).to have_content("can't be blank")
+    end
+  end
+
+  context 'when existing ingredient is submitted' do
```

Comment:
No need to test every possible case through integration test. You should write unit tests for your model to test such things. Basically In this integration test file you should cover only:

- creation succeeded
- creation failed becasue of validation errors.

Integration tests are generally slow.

Anyway we can leave it like it is.